ci/checks/run-cmake-format.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Copyright (c) 2021-2023, NVIDIA CORPORATION.
+
+# This script is a wrapper for cmakelang that may be used with pre-commit. The
```

Comment:
Should we use the upstream pre-commit hook (https://github.com/cheshirekow/cmake-format-precommit) instead of copying the `run-cmake-format.sh` file, since this repo contains the original config file and it doesn't need to be fetched?

Reply:
I am happy to defer this to a follow up PR

Reply:
I think we should switch, yes. The fetch was the only reason I implemented the script in the first place. Hopefully switching to the hook will also address the issue you observed with `--first-comment-is-literal`.

Reply:
I attempted to switch to the official pre-commit hook, but failed. There is no option to require `pre-commit` to run one file at a time, so the use of `--first-comment-is-literal` means we always run into the bug mentioned above. We have to use an intermediate shell script to split the work and execute `cmake-format` on individual files, so I guess we're stuck with using `run-cmake-format.sh`.